File: src/models/errors/session.ts
export class SuspendedIPError extends Error {
  constructor() {
    super("Your IP address has been suspended");
    this.name = "SuspendedIPError";
  }
}

export class SessionExpiredError extends Error {
  constructor() {
    super("The session has expired");
    this.name = "SessionExpiredError";
  }
}


export class PageUnavailableError extends Error {
  constructor() {
    super("The requested page does not exist");
    this.name = "PageUnavailableError";
  }
}

export class BusyPageError extends Error {
  constructor() {
    super("The site is temporarily unavailable");
    this.name = "BusyPageError";
  }
}

export class RateLimitedError extends Error {
  constructor() {
    super("You've been rate-limited");
    this.name = "RateLimitedError";
  }
}

export class ServerSideError extends Error {
  constructor(message = "An error occurred, server-side") {
    super(message);
    this.name = "ServerSideError";
  }
}

File: library/src/commonMain/kotlin/models/errors/session.kt
package models.errors

class SuspendedIPError: Exception("Your IP address has been suspended")

class SessionExpiredError: Exception("The session has expired")

class PageUnavailableError: Exception("The requested page does not exist")

class BusyPageError: Exception("The site is temporarily unavailable")

class RateLimitedError: Exception("You've been rate-limited")

class ServerSideError(message: String = "An error occurred, server-side"): Exception(message)